### DISCUSSION TITLE
Getting the HTML from a tiddler

### DISCUSSION THREAD

**Discussion by GameDungeon** [TiddlyWiki/TiddlyWiki5#7496] [Q&A] [answered]
In JS, how do I parse a tiddler and get the HTML from it? I need it as a string. I was thinking I could find something like it in the export code, but I actually can't find the export code.

**Comment by pmario** [comment, marked as answer]
The tiddler [$:/state/editpreviewtype](https://tiddlywiki.com/#%24%3A%2Fstate%2Feditpreviewtype) contains the preview type, that you can select in tiddler edit mode with the preview selector. 

![grafik](https://github.com/Jermolene/TiddlyWiki5/assets/374655/82e9a278-7391-46c9-ab86-050668826968)

It points to: [$:/plugins/tiddlywiki/internals/EditTemplate/body/preview/raw](https://tiddlywiki.com/#%24%3A%2Fplugins%2Ftiddlywiki%2Finternals%2FEditTemplate%2Fbody%2Fpreview%2Fraw), which contains the code, how it is implemented. 

See the follwing code. So it uses the [view-widget](https://tiddlywiki.com/#ViewWidget) with the `currentTiddler` as the default `tiddler` parameter. 

```
<pre><code><$view field="text" format="htmlwikified" /></code></pre>
```

The [wikify-widget](https://tiddlywiki.com/#WikifyWidget) has similar functionality and parameters. 

The JS function which can render a tiddler is [.renderTiddler = function(outputType,title,options) {](https://github.com/tavin/TiddlyWiki5/blob/04c40627a43378637bf4bcf86633e8defac9b7c5/core/modules/wiki.js#L1199) ... Also **have a closer look at the functions directly above this one**

- js code [view-widget](https://github.com/tavin/TiddlyWiki5/blob/04c40627a43378637bf4bcf86633e8defac9b7c5/core/modules/widgets/view.js#L6)
- js code [wikify-widget](https://github.com/tavin/TiddlyWiki5/blob/04c40627a43378637bf4bcf86633e8defac9b7c5/core/modules/widgets/wikify.js#L5)

IMO that should give you most of the pointers you'll need. 
Also have a look at: [UI and rendering pipeline](https://tiddlywiki.com/dev/#UI%20and%20Rendering%20Pipeline)
